packages/workbox-routing/Router.mjs

```diff
@@ -86,21 +85,27 @@ class Router {
    * ```
    */
   addCacheListener() {
-    self.addEventListener('message', (event) => {
-      const {type, meta, payload} = event.data;
+    self.addEventListener('message', async (event) => {
+      const {type, payload} = event.data;
 
-      if (type === 'CACHE_URLS' && meta === 'workbox-window') {
+      if (type === 'CACHE_URLS') {
         if (process.env.NODE_ENV !== 'production') {
           logger.debug(`Caching URLs from the window`, payload.urlsToCache);
         }
 
-        for (let entry of payload.urlsToCache) {
+        const requestPromises = payload.urlsToCache.map((entry) => {
           if (typeof entry === 'string') {
             entry = [entry];
           }
 
           const request = new Request(...entry);
-          this.handleRequest({request, event});
+          return this.handleRequest({request, event});
+        });
+
```

Comment:
I realize now that there should be an `event.waitUntil(requestPromises)` here (wrapped in a `try`/`catch` to handle browsers that don't support async `waitUntil`).

https://github.com/w3c/ServiceWorker/issues/669#issuecomment-402779578

Reply:
Nevermind! `handleRequest()` already calls `waitUntil()`.

Reply:
I believe all the built-in strategy handlers already add their own try/catch logic, so I don't think this is necessary here.